### PR TITLE
feat: governor spoke is able to execute multiple transactions atomically

### PR DESCRIPTION
**Pull request by mrice32**
**Motivation**

The `GovernorSpoke` contract should be able to execute multiple transactions atomically keep the Governance system from losing control of the sequencing of important actions.

**Summary**

This adds a more complex structure for cross-chain messages so that they can contain multiple transactions rather than one.

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [ ]  New unit tests created
- [x]  Existing tests adequate, no new tests required
- [ ]  All existing tests pass
- [ ]  Untested

**Issue(s)**

N/A

### CODE REVIEW

File: packages/core/contracts/cross-chain-oracle/GovernorSpoke.sol [hunk 1]
```diff
@@ -36,12 +41,15 @@ contract GovernorSpoke is Lockable, ChildMessengerConsumerInterface {
...
-        (address to, bytes memory inputData) = abi.decode(data, (address, bytes));
+        Call[] memory calls = abi.decode(data, (Call[]));
         // TODO: Consider calling this via <address>.call(): https://docs.soliditylang.org/en/v0.8.10/units-and-global-variables.html?highlight=low%20level%20call#members-of-address-types
         // to avoid inline assembly.
 
-        require(_executeCall(to, inputData), "execute call failed");
-        emit ExecutedGovernanceTransaction(to, inputData);
+        for (uint256 i = 0; i < calls.length; i++) {
+            (address to, bytes memory inputData) = (calls[i].to, calls[i].data);
+            require(_executeCall(to, inputData), "execute call failed");
```
Comment (chrismaree) [inline packages/core/contracts/cross-chain-oracle/GovernorSpoke.sol:50]: this forces that everything in the batch either success or fails, thereby making this atomic. nice.

File: packages/core/contracts/cross-chain-oracle/GovernorHub.sol [hunk 2]
```diff
@@ -41,17 +41,12 @@ contract GovernorHub is Ownable, Lockable, MultiCaller {
      * @notice This should be called in order to relay a governance request to the `GovernorSpoke` contract deployed to
      * the child chain associated with `chainId`.
      * @param chainId network that messenger contract will communicate with
-     * @param to Contract on child chain to send message to
-     * @param dataFromGovernor Message to send. Should contain the encoded function selector and params.
+     * @param calls the calls to be made by the GovernorSpoke.
```
Comment (chrismaree) [inline packages/core/contracts/cross-chain-oracle/GovernorHub.sol:44]:
```suggestion
     * @param calls the calls to be made by the GovernorSpoke. Should encode a `to` and `data` prop for each call.
```

File: packages/core/test/cross-chain-oracle/GovernorSpoke.js [hunk 3]
```diff
@@ -25,16 +28,37 @@ describe("GovernorSpoke.js", async () => {
...
     const setChildMessengerEvents = await governor.getPastEvents("SetChildMessenger", { fromBlock: 0 });
     assert.equal(setChildMessengerEvents.length, 1);
     assert.equal(setChildMessengerEvents[0].returnValues.childMessenger, messenger);
   });
+
   it("Can delegate call if called by Messenger", async function () {
-    let targetAddress = addressWhitelist.options.address;
-    let inputDataBytes = addressWhitelist.methods.addToWhitelist(messenger).encodeABI();
-    let messageBytes = web3.eth.abi.encodeParameters(["address", "bytes"], [targetAddress, inputDataBytes]);
+    const calls = [
```
Comment (chrismaree) [inline packages/core/test/cross-chain-oracle/GovernorSpoke.js:42]: this is really not too bad! having to do the `web3.eth.abi.encodeParameters` on this array of objects (struct) is a bit painful but nothing we cant abstract away in a script :)

**Review by chrismaree — APPROVED** [review]
this is awesome!

File: packages/core/contracts/cross-chain-oracle/GovernorSpoke.sol (hunk 1 above)
Comment (mrice32) [reply packages/core/contracts/cross-chain-oracle/GovernorSpoke.sol:50]: Yep!

**Pull request merged**